src/common/components/repositories-home/index.js

```diff
@@ -33,12 +36,22 @@ class RepositoriesHome extends Component {
 
   componentDidMount() {
     const { authenticated } = this.props.auth;
+    const { dispatch } = this.props;
     const owner = this.props.user.login;
 
     if (authenticated) {
-      this.props.dispatch(fetchUserSnaps(owner));
+      dispatch(fetchUserSnaps(owner));
+      if (!interval) {
+        interval = setInterval(() => {
+          dispatch(fetchUserSnaps(owner, false));
+        }, SNAP_POLL_PERIOD);
+      }
     }
+  }
 
+  componentWillUnmount() {
+    clearInterval(interval);
+    interval = null;
```

Comment:
Clear timer, preventing snap data being fetched when user navigates away from the page.